esp8266: Avoid including esp_mphal.h directly from source files

projectgus opened the pull request:
Noticed while doing another task, figured was worth cleaning up.

*This work was funded through GitHub Sponsors.*

dpgeorge commented:
This should not be needed, no other ports have this.

The idea is that this header is only ever included by `py/mphal.h`, and that file has the guards.

dpgeorge commented:
I think it should be fixed by including `py/mphal.h` instead.  Except for `esp_init_data.c` which just doesn't need to include it at all.

projectgus committed:
esp8266: Avoid including ep_mphal.h directly.

This header has no include guards and is apparently only supposed to be
included from py/mphal.h.

This work was funded through GitHub Sponsors.

Signed-off-by: Angus Gratton <angus@redyak.com.au>

projectgus commented:
> Maybe... but then strictly following that, mpconfigport.h and mpconfigboard.h need to be named that way, and they aren't really private or internal headers.

There could be ways around this ,though. All port-specific headers including these could go into a `port` subdirectories, for example. Or we could document somewhere that `mpconfig*.h` is a particular naming convention for those headers only.
